include/sundials/sundials_stepper.h

```diff
@@ -0,0 +1,113 @@
+                        N_Vector tmp3);
+
+typedef int (*SUNJacTimesFn)(N_Vector v, N_Vector Jv, sunrealtype t, N_Vector y,
+                             N_Vector fy, void* user_data, N_Vector tmp);
+
+typedef _SUNDIALS_STRUCT_ SUNStepper_* SUNStepper;
+
+typedef SUNErrCode (*SUNStepperEvolveFn)(SUNStepper stepper, sunrealtype t0,
+                                         sunrealtype tout, N_Vector vout,
+                                         sunrealtype* tret);
+
+typedef SUNErrCode (*SUNStepperOneStepFn)(SUNStepper stepper, sunrealtype t0,
+                                          sunrealtype tout, N_Vector vout,
+                                          sunrealtype* tret);
+
+typedef SUNErrCode (*SUNStepperFullRhsFn)(SUNStepper stepper, sunrealtype t,
+                                          N_Vector v, N_Vector f);
+
+typedef SUNErrCode (*SUNStepperResetFn)(SUNStepper stepper, sunrealtype tR,
+                                        N_Vector vR);
+
+typedef SUNErrCode (*SUNStepperSetStopTimeFn)(SUNStepper stepper,
+                                              sunrealtype tstop);
+
+typedef SUNErrCode (*SUNStepperSetForcingFn)(SUNStepper stepper,
+                                             sunrealtype tshift,
+                                             sunrealtype tscale,
+                                             N_Vector* forcing, int nforcing);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_Create(SUNContext sunctx, SUNStepper* stepper);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_Destroy(SUNStepper* stepper);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_Evolve(SUNStepper stepper, sunrealtype t0,
+                             sunrealtype tout, N_Vector vout, sunrealtype* tret);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_OneStep(SUNStepper stepper, sunrealtype t0,
+                              sunrealtype tout, N_Vector vout, sunrealtype* tret);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_Reset(SUNStepper stepper, sunrealtype tR, N_Vector vR);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_SetStopTime(SUNStepper stepper, sunrealtype tstop);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_SetForcing(SUNStepper stepper, sunrealtype tshift,
+                                 sunrealtype tscale, N_Vector* forcing,
+                                 int nforcing);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_SetContent(SUNStepper stepper, void* content);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_GetContent(SUNStepper stepper, void** content);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_SetLastFlag(SUNStepper stepper, int last_flag);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_GetLastFlag(SUNStepper stepper, int* last_flag);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_SetEvolveFn(SUNStepper stepper, SUNStepperEvolveFn fn);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_SetOneStepFn(SUNStepper stepper, SUNStepperOneStepFn fn);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNStepper_SetFullRhsFn(SUNStepper stepper, SUNStepperFullRhsFn fn);
```

Comment:
Note to reviewers: there is no analogous `SUNStepper_FullRhs`. This is following what MRI does where it was presumably excluded because the full RHS functions are private in ARKODE. I can make a public `SUNStepper_FullRhs` function if folks want.

Reply:
I think this could make sense, especially since something using a `SUNStepper` might need the full RHS for some reason